extendr-api/src/robj/into_robj.rs

```diff
@@ -426,6 +426,103 @@ macro_rules! impl_str_tvv {
 impl_str_tvv! {&str}
 impl_str_tvv! {String}
 
+impl ToVectorValue for Rstr {
+    fn sexptype() -> SEXPTYPE {
+        SEXPTYPE::STRSXP
+    }
+
+    fn to_sexp(&self) -> SEXP
+    where
+        Self: Sized,
+    {
+        unsafe { self.get() }
+    }
+}
+
+impl ToVectorValue for &Rstr {
+    fn sexptype() -> SEXPTYPE {
+        SEXPTYPE::STRSXP
+    }
+
+    fn to_sexp(&self) -> SEXP
+    where
+        Self: Sized,
+    {
+        unsafe { self.get() }
+    }
+}
+
+impl ToVectorValue for Option<Rstr> {
+    fn sexptype() -> SEXPTYPE {
+        SEXPTYPE::STRSXP
+    }
+
+    fn to_sexp(&self) -> SEXP
+    where
+        Self: Sized,
+    {
+        if let Some(s) = self {
+            unsafe { s.get() }
+        } else {
+            unsafe { R_NaString }
+        }
+    }
+}
+
+impl TryFrom<&Robj> for Rstr {
+    type Error = crate::Error;
+
+    fn try_from(robj: &Robj) -> Result<Self> {
+        let sexptype = robj.sexptype();
+        if let SEXPTYPE::STRSXP = sexptype {
+            if robj.len() == 1 {
+                let strs = Strings::try_from(robj)?;
+                Ok(strs.elt(0))
+            } else {
+                Err(Error::ExpectedRstr(robj.clone()))
+            }
+        } else if let SEXPTYPE::CHARSXP = sexptype {
+            Ok(Rstr { robj: robj.clone() })
+        } else {
+            Err(Error::ExpectedRstr(robj.clone()))
+        }
+    }
+}
+
+impl TryFrom<Robj> for Rstr {
+    type Error = crate::Error;
+
+    fn try_from(value: Robj) -> std::result::Result<Self, Self::Error> {
+        Self::try_from(&value)
+    }
+}
+
+impl GetSexp for Rstr {
+    unsafe fn get(&self) -> SEXP {
+        self.robj.get()
+    }
+
+    unsafe fn get_mut(&mut self) -> SEXP {
+        self.robj.get_mut()
+    }
+
+    fn as_robj(&self) -> &Robj {
+        &self.robj
+    }
+
+    fn as_robj_mut(&mut self) -> &mut Robj {
+        &mut self.robj
+    }
+}
+
+// These traits all derive from GetSexp with default implementations
+impl Length for Rstr {}
+impl Types for Rstr {}
+impl Conversions for Rstr {}
+impl Rinternals for Rstr {}
+impl Slices for Rstr {}
+impl Operators for Rstr {}
```

Comment:
this frustrated me lol! Why don't we have blanket impls for these?

Reply:
I did this once.. and it was glorious, but that branch didn't make it to PRs.